src/systemTest/java/com/flightstats/hub/model/ChannelConfigExpirationSettings.java

```diff
@@ -0,0 +1,73 @@
+package com.flightstats.hub.model;
+
+import lombok.Builder;
+import lombok.Value;
+import org.joda.time.DateTime;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Builder
+@Value
+public class ChannelConfigExpirationSettings {
+    String channelName;
+
+    boolean keepForever;
+    long ttlDays;
+    DateTime mutableTime;
+    long maxItems;
+
+    public Optional<DateTime> getMutableTime() {
+        return Optional.ofNullable(mutableTime);
+    }
+
+    public static class ChannelConfigExpirationSettingsBuilder {
+        private boolean keepForever;
+        private long ttlDays;
+        private DateTime mutableTime;
+        private long maxItems;
+
+        public ChannelConfigExpirationSettingsBuilder keepForever(boolean keepForever) {
+            this.keepForever = keepForever;
+            validate();
```

Comment:
I'd think the validate would be at the ```.build()``` time rather than each setter. That also allows it to do validation for combinations (e.g. a && b must be set) that wouldn't be possible at individual setter time.

Also, less code, you'd only override build() rather than every setter.

Reply:
I was worried `build()` wouldn't work right if I defined it here, but that worry is probably silly. I'll move it.